src/docs/arch-column-row.dox

```diff
@@ -0,0 +1,99 @@
+/*! @arch_page arch-column-row Row Store & Column Store
+
+@section column_row_definition Definition
+
+A row oriented storage is an access method where all columns of a row are stored
+together, the data is stored by row, while a column oriented system stores its data tables by
+column. WiredTiger supports both and each management system is represented internally by a certain
+type of B-Tree data structure.
+
+Keys and values in row-stores can be of variable length, thus, row-stores implement a generalized
+version of a B-Tree. Furthermore, keys are explicitly stored. Column-stores use a 64-bit unsigned
+integer record id as keys, thus, column-stores implement a specialized version of a B-Tree where the
+key has a predictable length. In the case of column-stores, keys are implied given the row's
+position in the B-Tree.
+
+There are two types of column-stores in WiredTiger:
+- The fixed length column-store, as its name implies, the value has a fixed length, and furthermore
+the value is restricted to between 1 and 8 bits in length. The bit length is specified when the
+column store is created. The fixed length column store has specialized use cases like bitmaps.
+- The variable length is a more general case which allows for values to have any length similarly to
+row-stores.
+
+Please refer to @ref arch-data-file to learn more about the on-disk format for row-stores and
+column-stores.
+
+@section column_row_rs Row-stores
+
+Row-stores are probably seen as the traditional access method and they serve a general purpose. In
+WiredTiger, row-store keys are explicitly stored and they can be duplicated within the same single
+file which directly impacts the on-disk storage size. However, keys are only duplicated if they are
+the first key on the page, which gets them copied/promoted to an internal page. They are not
+necessarily duplicated, too, they can be prefix/suffix compressed to discard unnecessary bytes on
+the internal pages.
+
+@section column_row_vlcs Variable length column-stores
+
+The main advantage of variable length column-stores is on-disk storage size. Their keys are not
```

Comment:
Isn't the main advantage that the cache doesn't necessarily load columns that aren't needed for the query? (That does start to venture into performance, based on @ddanderson's comments, maybe we just don't talk about "advantages" in this page?)

Reply:
I guess those two are one of the CS advantages. I remember having a discussion with @agorrod saying that saving disk space was definitely one of the main advantages of the CS, I may have misunderstood.
I could effectively remove any "advantages" since it seems that is should not be part of this page.

Reply:
See fd5ffdacd933d6f5099c3253dbc586de30c43d47